src/TTimer.h

```diff
@@ -46,17 +47,17 @@ class TTimer : public Tree<TTimer>
     TTimer(TTimer* parent, Host* pHost);
     TTimer(const QString& name, QTime time, Host* pHost, bool repeating = false);
     void compileAll();
-    QString& getName() { return mName; }
+    const QString& getName() const { return mName; }
     void setName(const QString& name);
-    QTime& getTime() { return mTime; }
+    const QTime& getTime() const { return mTime; }
     void compile();
     bool checkRestart();
     bool compileScript();
     void execute();
     void setTime(QTime time);
-    QString getCommand() { return mCommand; }
+    const QString getCommand() const { return mCommand; }
```

Comment:
The `const` is useless unless you are returning reference rather than a copy remember...
```suggestion
    const QString& getCommand() const { return mCommand; }
```

Reply:
So wouldn't your suggestion make the const useless? Qt classes are also CoW, so there's no real need to return them as a reference. The current code we have works ok

Reply:
IIRC Clazy (or possibly Qt Creator's own) code analysis will point out that `const` on the return value of a method that returns a value rather than a reference is useless/pointless as it has no effect. If the return is a reference than it ***is*** significant. The `const` after the method declaration but before the definition is a different matter and it is of use to prevent that the class members being modified (and signify that that is the case) for that method  (unless they have been declared `mutable`!)

Basically, the leftmost `const` is not needed here if we are not going to change to return a reference...

Reply:
That makes sense :)

Applied.